File: backend/app/schemas/tax.py
from pydantic import BaseModel


class TaxRequest(BaseModel):
    purchase_price: float
    citizenship: str  # SC | PR | Foreigner | Entity
    existing_property_count: int
    renovation_budget: float = 0.0
    agent_rate: float = 0.01


class TaxResponse(BaseModel):
    purchase_price: float
    bsd: float
    absd: float
    legal_fees: float
    agent_commission: float
    valuation_fee: float
    renovation_budget: float
    total_cash_outlay: float


class SSDRequest(BaseModel):
    purchase_price: float
    hold_years: float


class SSDResponse(BaseModel):
    ssd: float
    ssd_rate: float
    hold_years: float

File: frontend/src/api/tax.ts
import { apiPost } from './client'

export interface TaxRequest {
  purchase_price: number
  citizenship: string
  existing_property_count: number
  renovation_budget?: number
  agent_rate?: number
}

export interface TaxResponse {
  purchase_price: number
  bsd: number
  absd: number
  legal_fees: number
  agent_commission: number
  valuation_fee: number
  renovation_budget: number
  total_cash_outlay: number
}

export interface SSDRequest {
  purchase_price: number
  hold_years: number
}

export interface SSDResponse {
  ssd: number
  ssd_rate: number
  hold_years: number
}

export const calculateUpfrontCost = (req: TaxRequest) =>
  apiPost<TaxResponse>('/api/tax/upfront-cost', req)

export const calculateSsd = (req: SSDRequest) =>
  apiPost<SSDResponse>('/api/tax/ssd', req)
